test/Makefile

```diff
@@ -26,6 +26,12 @@ all:
 check: $(EXE)
 	./run_test.sh
 
+valgrind-check: $(EXE)
+ ifeq (, $(shell which valgrind))
+ $(error "No valgrind found in $(PATH), skipping valgrind tests")
+ endif
+	TEST_WRAPPER="valgrind --leak-check=full --suppressions=valgrind.supp" ./run_test.sh
```

Comment:
I was thinking more about this and I'm afraid this is going to cause trouble in the future, e.g. if we decide to stop running tests serially and start running them in parallel, this recipe would have to be rewritten.
I believe we can still provide a `valgrind-check` target in the main Makefile (this patch already does that), but running:
`$(MAKE) -C test check TEST_WRAPPER="valgrind --leak-check=full --suppressions=valgrind.supp"`

I also suggest to remove the test for valgrind in this recipe.